pkg/store/prometheus.go

```diff
@@ -537,10 +533,11 @@ func (p *PrometheusStore) encodeChunk(ss []prompb.Sample) (storepb.Chunk_Encodin
 func (p *PrometheusStore) LabelNames(ctx context.Context, r *storepb.LabelNamesRequest) (*storepb.LabelNamesResponse, error) {
 	extLset := p.externalLabelsFn()
 
-	match, matchers, err := matchesExternalLabels(r.Matchers, extLset)
```

Comment:
prefer not  change the interface and ease the effort to resolve upstream conflicts

Reply:
ack